### PR TITLE
Malformed file detection script

### PR DESCRIPTION
**Pull request by isobarbaric**

### CODE REVIEW

File: src/mdframe/reader.py [hunk 1]
```diff
@@ -27,19 +31,20 @@ def metadata_file_to_df(metadata_file_contents):
...
     return list(metadata_records)
 
-
 @dataclass
 class Config:
     data_dir_path: Path
     metadata_file_extension: Optional[MetadataFileExtension]
     # data_file_extensions: List[DataFileExtension]
-
+    schema: Dict
```
Comment (kjczarne) [inline src/mdframe/reader.py:46]: You can use a `Dict` here but I would suggest the `schema.py` to be actually dumped into JSON. In this format we can leverage JSONSchema plugins in e.g. VSCode. As for how the dict would be loaded here you could do something like:

```python
from pathlib import Path
from urllib3 import Url


@dataclass
class Config:
    data_dir_path: Path
    metadata_file_extension: Optional[MetadataFileExtension]
    # data_file_extensions: List[DataFileExtension]
    schema_url: Url | Path
    __schema: Dict | None = None

    @property
    def schema(self):
        if self.__schema is None:
            self.__schema = # code to load dict schema from a JSON file
        return self.__schema
```

File: src/mdframe/reader.py [hunk 2]
```diff
@@ -57,8 +62,9 @@ def main():
                         default="toml")
     args = parser.parse_args()
     config = Config(
-        Path(args.directory),
-        args.metadata_ext,
+        data_dir_path = Path(args.directory),
```
Comment (kjczarne) [inline src/mdframe/reader.py:65]: Good usage of kwargs. One nitpick, when passing kwargs you should skip the space around `=`, so e.g. `data_dir_path=Path(args.directory)`. Elsewhere, you include whitespace around operators.

File: src/mdframe/schema.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
+schema = {
+  "$schema": "http://json-schema.org/draft-04/schema#",
```
Comment (kjczarne) [inline src/mdframe/schema.py:2]: Like mentioned in my other comment you might want to store this as an actual JSON file committed into the repository and read that JSON when the `Config` instance is being constructed.

File: test/data_dir/data_file3.toml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+gid = 75
```
Comment (kjczarne) [inline test/data_dir/data_file3.toml:1]: Thanks for adding full files for testing, this is going to be very useful to develop further tests.

**Review by kjczarne — CHANGES_REQUESTED** [review]
Great job! One thing to change would be to use an actual JSON file instead of a Python dict to distribute the schema.

File: src/mdframe/reader.py (hunk 2 above)
Comment (isobarbaric) [reply src/mdframe/reader.py:65]: taken care of :thumbsup:

File: src/mdframe/reader.py (hunk 1 above)
Comment (isobarbaric) [reply src/mdframe/reader.py:46]: taken care of (distinguished between Url, Path cases)

File: src/mdframe/schema.py (hunk 3 above)
Comment (isobarbaric) [reply src/mdframe/schema.py:2]: done 🎸

File: test/data_dir/data_file3.toml (hunk 4 above)
Comment (isobarbaric) [reply test/data_dir/data_file3.toml:1]: 🚀

**Review by kjczarne — APPROVED** [review]
Looks great! I'm accepting this.

**Pull request merged**